app/src/main/java/com/example/authapp/social/FacebookLoginProcedure.kt

```diff
@@ -1,87 +1,63 @@
 package com.example.authapp.social
 
 import android.app.Activity
-import android.content.Context
 import android.content.Intent
-import android.util.Log
-import android.view.View
-import android.widget.ProgressBar
-import android.widget.Toast
 import com.example.authapp.R
-import com.example.authapp.handler.PreferenceHandler
-import com.example.authapp.ui.UserDashboardActivity
-import com.facebook.AccessToken
-import com.facebook.CallbackManager
-import com.facebook.FacebookCallback
-import com.facebook.FacebookException
+import com.example.authapp.callback.StateChangesListener
+import com.example.authapp.status.LoginStatus
+import com.facebook.*
 import com.facebook.login.LoginManager
 import com.facebook.login.LoginResult
 import com.google.firebase.auth.FacebookAuthProvider
 import com.google.firebase.auth.FirebaseAuth
-import com.google.firebase.auth.FirebaseUser
 
-class FacebookLoginProcedure(private val context: Context) {
+class FacebookLoginProcedure {
 
     private val managerCallback : CallbackManager by lazy { CallbackManager.Factory.create() }
     private val firebaseAuth : FirebaseAuth by lazy { FirebaseAuth.getInstance() }
-    private val readPermissions : Array<String> by lazy { context.resources.getStringArray(R.array.read_permissions) }
+    private lateinit var readPermissions : Array<String>
+    private lateinit var stateChangesListener : StateChangesListener
 
-    fun setUpFacebookLogin(activity : Activity, progressBar: ProgressBar) {
+    fun getStateListener(listener : StateChangesListener) {
+        stateChangesListener = listener
+    }
+
+    fun startFacebookLogin(activity : Activity) {
+        readPermissions = activity.resources.getStringArray(R.array.read_permissions)
```

Comment:
Get this permission as an array from the activity, instead of retrieving it here.